app/Models/Image.php

```diff
@@ -14,4 +14,8 @@ class Image extends Model
         'object-type',
         'object-id',
     ];
+    public function tour()
+    {
+        return $this->belongsTo(Tour::class, 'object-id', 'id');
```

Comment:
Table images này tụi e muốn dùng polymorphic à em?
Hay sao mà có object-type với object-id?

Reply:
dạ anh

Reply:
sao ko dùng chuẩn của laravel cho tiện em?

Reply:
Nếu dùng polynophic thì khao báo relation dùng morphTo và morphMany chứ sao lại dùng belongsTo với hasMany thì có khác gì quan hệ One-Many bình thường đâu e?
https://laravel.com/docs/8.x/eloquent-relationships#one-to-many-polymorphic-relations
Hơn nữa nếu dùng polynophic nên đặt tên dạng imageable. 2 trường xác định object nó thuộc về nên đặt là  imagable_id và imageable_type theo chuẩn của laravel suggestion

Reply:
> Nếu dùng polynophic thì khao báo relation dùng morphTo và morphMany chứ sao lại dùng belongsTo với hasMany thì có khác gì quan hệ One-Many bình thường đâu e?
> https://laravel.com/docs/8.x/eloquent-relationships#one-to-many-polymorphic-relations
> Hơn nữa nếu dùng polynophic nên đặt tên dạng imageable. 2 trường xác định object nó thuộc về nên đặt là imagable_id và imageable_type theo chuẩn của laravel suggestion

Dạ vậy có gì nhóm em sửa lại rồi pull ạ.

Reply:
Ok, có thể chấp nhận cho làm đến khúc đó nhận ra sai rồi sửa.
https://github.com/awesome-academy/php_oe45_booking/pull/12/files#diff-f218fd6cd4e15c7e12970aa62347f2c9a52185bb673d8c97274c0035c911b70dR19
Nhưng lưu ý không như pull trước đó anh Quốc có mợt nhầm

Bất kỳ sự thay đổi nào liên quan đến database như thêm column, sửa column, type, của 1 table trước đó các bạn đều phải tạo mới 1 file migration dành riêng cho việc đó . Sau đó chạy migrate để nó cập nhật theo file migration mới. Giữ nguyên data hiện tại
Không sửa trực tiếp vào file cũ như này. Không thôi không phải drop database rồi create mới từ đầu. 1 dự án mà clean DB về rỗng hết thế này ? Chưa kể mỗi người trong team đều phải lặp lại thao tác đó!

Reply:
> Ok, có thể chấp nhận cho làm đến khúc đó nhận ra sai rồi sửa.
> https://github.com/awesome-academy/php_oe45_booking/pull/12/files#diff-f218fd6cd4e15c7e12970aa62347f2c9a52185bb673d8c97274c0035c911b70dR19
> Nhưng lưu ý không như pull trước đó anh Quốc có mợt nhầm
> 
> Bất kỳ sự thay đổi nào liên quan đến database như thêm column, sửa column, type, của 1 table trước đó các bạn đều phải tạo mới 1 file migration dành riêng cho việc đó . Sau đó chạy migrate để nó cập nhật theo file migration mới. Giữ nguyên data hiện tại
> Không sửa trực tiếp vào file cũ như này. Không thôi không phải drop database rồi create mới từ đầu. 1 dự án mà clean DB về rỗng hết thế này ? Chưa kể mỗi người trong team đều phải lặp lại thao tác đó!

dạ a.